index.gohtml

```diff
@@ -1,15 +1,90 @@
 <!DOCTYPE html>
-<html lang="en">
+<html>
 <head>
-    <meta charset="UTF-8">
-    <title>System Monitor</title>
+    <title>Resource Tracker</title>
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
+    <script src="script.js"></script>
 </head>
 <body>
-<h1>System Monitor</h1>
-<p>CPU Utilization: {{index .CpuPercent 0}}%</p>
-<p>Memory Utilization: {{.MemPercent.UsedPercent}}%</p>
-{{if .Message}}
-    <p style="color:red;">{{.Message}}</p>
-{{end}}
+<div class="container">
+    <h1>Resource Tracker</h1>
+    <div id="cpu-mem-timeseries"></div>
+</div>
+<script>
+    // Initialize arrays to store the data
+    var timestamps = [];
+    var cpuUsage = [];
+    var memUsage = [];
+
+    // Function to get the current time in a suitable format
+    function getCurrentTime() {
+        return new Date().toISOString();
+    }
+
+    // Function to fetch data from the server
+    function fetchData() {
+        $.ajax({
+            url: 'http://localhost:8080/usage',  // Updated port number
+            method: 'GET',
+            success: function(data) {
+                updateGraph(data.cpuPercent, data.memPercent);
+            }
+        });
+    }
```

Comment:
Can add error handling to the (`$.ajax`) call to handle cases where the server is unreachable or returns error

```suggestion
function fetchData() {
    $.ajax({
        url: 'http://localhost:8080/usage',
        method: 'GET',
        success: function(data) {
            updateGraph(data.cpuPercent, data.memPercent);
        },
        error: function(xhr, status, error) {
            // Handle errors here
            console.error('Error fetching data:', error);
            // Optionally, you can display an error message to the user
            alert('Error fetching data. Please try again later.');
        }
    });
}

```